samcli/commands/pipeline/init/pipeline_templates_manifest.py

```diff
@@ -0,0 +1,50 @@
+"""
+Represents a manifest that lists the available SAM pipeline templates.
+Example:
+    providers:
```

Comment:
`templates` already has identical `provider`s. Why is another `providers` necessary?

Reply:
We can extract the unique providers from the templates metadata, but defining them separately comes with two benefits:
1. readability; you instantly figure out which providers we support
2. consistent user experience; In this approach, this list of providers will be displayed to the user in this same order every time they run the command. The other approach doesn't guarantee this.

Reply:
For (2), `templates` is a list so each `provider` will also be ordered, no?

Reply:
it depends on how we order the templates. Let's say we order them first by number of stages (single-stage-pipelines, two-stages-pipelines...) then by provider: Jenkins, Gitlab, GithubActions

For example, Initially we may have:
```
templates:
  - single-stage-gitlab
  - single-stage-github-actions
  - two-stages-jenkins
  - two-stages-gitlab
  - two-stages-github-actions
```
which gave `["gitlab", "github-actions", "jenkins"]`
Now the single-stage-jenkins pipeline template became ready and we added it to its order on top of the list, then the providers list will change accordingly to ` [ "jenkins", "gitlab", "github-actions"]`

Reply:
Okay I see, so it's essentially to ensure prompt ordering.

Is `provider`(`s`) internal identifiers or human-readable names? If the latter, they should be `GitLab CI/CD` and `GitHub Actions`.